crop_classification/torchserve_handler.py

```diff
@@ -0,0 +1,99 @@
+import sys
+import tempfile
+import time
+
+from pathlib import Path
+from google.cloud import storage  # type: ignore
+from ts.torch_handler.base_handler import BaseHandler
+from typing import Tuple
+
+from cropharvest.inference import Inference
+
+temp_dir = tempfile.gettempdir()
+dest_bucket_name = "crop-type-preds"
+
+
+class ModelHandler(BaseHandler):
+    """
+    A custom model handler implementation.
+    The basehandler calls initialize() on server start up and
+    preprocess(), inference(), and postprocess() on each request.
+    """
+
+    def __init__(self):
+        print("HANDLER: Starting up handler")
+        super().__init__()
+
+    def download_file(self, uri: str):
+        uri_as_path = Path(uri)
+        bucket_name = uri_as_path.parts[1]
+        file_name = "/".join(uri_as_path.parts[2:])
+        bucket = storage.Client().bucket(bucket_name)
+        retries = 3
+        blob = bucket.blob(file_name)
+        for i in range(retries + 1):
+            if blob.exists():
+                print(f"HANDLER: Verified {uri} exists.")
+                break
+            if i == retries:
+                raise ValueError(f"HANDLER ERROR: {uri} does not exist.")
+
+            print(
+                f"HANDLER: {uri} does not yet exist, sleeping for 5 seconds and trying again."
+            )
+            time.sleep(5)
+        local_path = f"{tempfile.gettempdir()}/{uri_as_path.name}"
+        blob.download_to_filename(local_path)
+        if not Path(local_path).exists():
+            raise FileExistsError(f"HANDLER: {uri} from storage was not downloaded")
+        print(f"HANDLER: Verified file downloaded to {local_path}")
+        return local_path
+
+    def initialize(self, context):
+        print("HANDLER: Initializing handler")
+        super().initialize(context)
+        properties = context.system_properties
+        model_dir = properties.get("model_dir")
+        sys.path.append(model_dir)
+
+        # TODO need to pass the normalizing dict here somehow
+        print("HANDLER: Initializing inference module")
+        self.inference_module = Inference(model=self.model, normalizing_dict=None)
```

Comment:
@gabrieltseng The easiest way that I have found to pass the normalizing_dict here is by making it an attribute of the model. (Note jit can only have primitive attributes so normalizing dict must be `Dict[str, List[float]]` instead of `Dict[str, List[np.ndarray]]`

I think this would make sense to set when running the fine-tuning script described above. Thoughts? We can discuss on Friday too